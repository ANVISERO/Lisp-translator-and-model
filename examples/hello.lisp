( defvar h " h " )
( defvar e " e " )
( defvar l " l " )
( defvar o " o " )
( defvar tab " " )
( defvar w " w " )
( defvar r " r " )
( defvar d " d " )

( format f h )
( format f e )
( format f l )
( format f l )
( format f o )
( format f tab )
( format f w )
( format f o )
( format f r )
( format f l )
( format f d )