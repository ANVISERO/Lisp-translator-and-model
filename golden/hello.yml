in_source: |-
  (
      ( defvar h " h " )
      ( defvar e " e " )
      ( defvar l " l " )
      ( defvar o " o " )
      ( defvar tab " " )
      ( defvar w " w " )
      ( defvar r " r " )
      ( defvar d " d " )

      ( print h )
      ( print e )
      ( print l )
      ( print l )
      ( print o )
      ( print tab )
      ( print w )
      ( print o )
      ( print r )
      ( print l )
      ( print d )
  )
in_stdin: |-
  foo
out_code: |-
  [{"opcode": "movv", "arg": ["$0", "h"]},
   {"opcode": "movv", "arg": ["$1", "e"]},
   {"opcode": "movv", "arg": ["$2", "l"]},
   {"opcode": "movv", "arg": ["$3", "o"]},
   {"opcode": "movv", "arg": ["$4", " "]},
   {"opcode": "movv", "arg": ["$5", "w"]},
   {"opcode": "movv", "arg": ["$6", "r"]},
   {"opcode": "movv", "arg": ["$7", "d"]},
   {"opcode": "movv", "arg": ["$69", "$0"]},
   {"opcode": "movv", "arg": ["$69", "$1"]},
   {"opcode": "movv", "arg": ["$69", "$2"]},
   {"opcode": "movv", "arg": ["$69", "$2"]},
   {"opcode": "movv", "arg": ["$69", "$3"]},
   {"opcode": "movv", "arg": ["$69", "$4"]},
   {"opcode": "movv", "arg": ["$69", "$5"]},
   {"opcode": "movv", "arg": ["$69", "$3"]},
   {"opcode": "movv", "arg": ["$69", "$6"]},
   {"opcode": "movv", "arg": ["$69", "$2"]},
   {"opcode": "movv", "arg": ["$69", "$7"]},
   {"opcode": "halt"}]
out_stdout: |
  source LoC: 101 code instr: 20
  ============================================================
  hello world
  instr_counter: 19  ticks: 106
out_log: |
  DEBUG   machine:simulation    TICK: 0, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0 movv ['$0', 'h']
  DEBUG   machine:tick          TICK: 1, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 0 movv ['$0', 'h']
  DEBUG   machine:tick          TICK: 2, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 0, DR: 104 movv ['$0', 'h']
  DEBUG   machine:tick          TICK: 3, PC: 0, ADDR: 0, MEM_OUT: 0, AC: 104, DR: 104 movv ['$0', 'h']
  DEBUG   machine:write         input: 'h'
  DEBUG   machine:tick          TICK: 4, PC: 0, ADDR: 0, MEM_OUT: 104, AC: 104, DR: 104 movv ['$0', 'h']
  DEBUG   machine:tick          TICK: 5, PC: 1, ADDR: 0, MEM_OUT: 104, AC: 104, DR: 104 movv ['$1', 'e']
  DEBUG   machine:tick          TICK: 6, PC: 1, ADDR: 1, MEM_OUT: 0, AC: 104, DR: 104 movv ['$1', 'e']
  DEBUG   machine:tick          TICK: 7, PC: 1, ADDR: 1, MEM_OUT: 0, AC: 104, DR: 101 movv ['$1', 'e']
  DEBUG   machine:tick          TICK: 8, PC: 1, ADDR: 1, MEM_OUT: 0, AC: 101, DR: 101 movv ['$1', 'e']
  DEBUG   machine:write         input: 'e'
  DEBUG   machine:tick          TICK: 9, PC: 1, ADDR: 1, MEM_OUT: 101, AC: 101, DR: 101 movv ['$1', 'e']
  DEBUG   machine:tick          TICK: 10, PC: 2, ADDR: 1, MEM_OUT: 101, AC: 101, DR: 101 movv ['$2', 'l']
  DEBUG   machine:tick          TICK: 11, PC: 2, ADDR: 2, MEM_OUT: 0, AC: 101, DR: 101 movv ['$2', 'l']
  DEBUG   machine:tick          TICK: 12, PC: 2, ADDR: 2, MEM_OUT: 0, AC: 101, DR: 108 movv ['$2', 'l']
  DEBUG   machine:tick          TICK: 13, PC: 2, ADDR: 2, MEM_OUT: 0, AC: 108, DR: 108 movv ['$2', 'l']
  DEBUG   machine:write         input: 'l'
  DEBUG   machine:tick          TICK: 14, PC: 2, ADDR: 2, MEM_OUT: 108, AC: 108, DR: 108 movv ['$2', 'l']
  DEBUG   machine:tick          TICK: 15, PC: 3, ADDR: 2, MEM_OUT: 108, AC: 108, DR: 108 movv ['$3', 'o']
  DEBUG   machine:tick          TICK: 16, PC: 3, ADDR: 3, MEM_OUT: 0, AC: 108, DR: 108 movv ['$3', 'o']
  DEBUG   machine:tick          TICK: 17, PC: 3, ADDR: 3, MEM_OUT: 0, AC: 108, DR: 111 movv ['$3', 'o']
  DEBUG   machine:tick          TICK: 18, PC: 3, ADDR: 3, MEM_OUT: 0, AC: 111, DR: 111 movv ['$3', 'o']
  DEBUG   machine:write         input: 'o'
  DEBUG   machine:tick          TICK: 19, PC: 3, ADDR: 3, MEM_OUT: 111, AC: 111, DR: 111 movv ['$3', 'o']
  DEBUG   machine:tick          TICK: 20, PC: 4, ADDR: 3, MEM_OUT: 111, AC: 111, DR: 111 movv ['$4', ' ']
  DEBUG   machine:tick          TICK: 21, PC: 4, ADDR: 4, MEM_OUT: 0, AC: 111, DR: 111 movv ['$4', ' ']
  DEBUG   machine:tick          TICK: 22, PC: 4, ADDR: 4, MEM_OUT: 0, AC: 111, DR: 32 movv ['$4', ' ']
  DEBUG   machine:tick          TICK: 23, PC: 4, ADDR: 4, MEM_OUT: 0, AC: 32, DR: 32 movv ['$4', ' ']
  DEBUG   machine:write         input: ' '
  DEBUG   machine:tick          TICK: 24, PC: 4, ADDR: 4, MEM_OUT: 32, AC: 32, DR: 32 movv ['$4', ' ']
  DEBUG   machine:tick          TICK: 25, PC: 5, ADDR: 4, MEM_OUT: 32, AC: 32, DR: 32 movv ['$5', 'w']
  DEBUG   machine:tick          TICK: 26, PC: 5, ADDR: 5, MEM_OUT: 0, AC: 32, DR: 32 movv ['$5', 'w']
  DEBUG   machine:tick          TICK: 27, PC: 5, ADDR: 5, MEM_OUT: 0, AC: 32, DR: 119 movv ['$5', 'w']
  DEBUG   machine:tick          TICK: 28, PC: 5, ADDR: 5, MEM_OUT: 0, AC: 119, DR: 119 movv ['$5', 'w']
  DEBUG   machine:write         input: 'w'
  DEBUG   machine:tick          TICK: 29, PC: 5, ADDR: 5, MEM_OUT: 119, AC: 119, DR: 119 movv ['$5', 'w']
  DEBUG   machine:tick          TICK: 30, PC: 6, ADDR: 5, MEM_OUT: 119, AC: 119, DR: 119 movv ['$6', 'r']
  DEBUG   machine:tick          TICK: 31, PC: 6, ADDR: 6, MEM_OUT: 0, AC: 119, DR: 119 movv ['$6', 'r']
  DEBUG   machine:tick          TICK: 32, PC: 6, ADDR: 6, MEM_OUT: 0, AC: 119, DR: 114 movv ['$6', 'r']
  DEBUG   machine:tick          TICK: 33, PC: 6, ADDR: 6, MEM_OUT: 0, AC: 114, DR: 114 movv ['$6', 'r']
  DEBUG   machine:write         input: 'r'
  DEBUG   machine:tick          TICK: 34, PC: 6, ADDR: 6, MEM_OUT: 114, AC: 114, DR: 114 movv ['$6', 'r']
  DEBUG   machine:tick          TICK: 35, PC: 7, ADDR: 6, MEM_OUT: 114, AC: 114, DR: 114 movv ['$7', 'd']
  DEBUG   machine:tick          TICK: 36, PC: 7, ADDR: 7, MEM_OUT: 0, AC: 114, DR: 114 movv ['$7', 'd']
  DEBUG   machine:tick          TICK: 37, PC: 7, ADDR: 7, MEM_OUT: 0, AC: 114, DR: 100 movv ['$7', 'd']
  DEBUG   machine:tick          TICK: 38, PC: 7, ADDR: 7, MEM_OUT: 0, AC: 100, DR: 100 movv ['$7', 'd']
  DEBUG   machine:write         input: 'd'
  DEBUG   machine:tick          TICK: 39, PC: 7, ADDR: 7, MEM_OUT: 100, AC: 100, DR: 100 movv ['$7', 'd']
  DEBUG   machine:tick          TICK: 40, PC: 8, ADDR: 7, MEM_OUT: 100, AC: 100, DR: 100 movv ['$69', '$0']
  DEBUG   machine:tick          TICK: 41, PC: 8, ADDR: 0, MEM_OUT: 104, AC: 100, DR: 100 movv ['$69', '$0']
  DEBUG   machine:tick          TICK: 42, PC: 8, ADDR: 0, MEM_OUT: 104, AC: 100, DR: 104 movv ['$69', '$0']
  DEBUG   machine:tick          TICK: 43, PC: 8, ADDR: 69, MEM_OUT: 0, AC: 100, DR: 104 movv ['$69', '$0']
  DEBUG   machine:tick          TICK: 44, PC: 8, ADDR: 69, MEM_OUT: 0, AC: 104, DR: 104 movv ['$69', '$0']
  DEBUG   machine:write         output: '' << 'h'
  DEBUG   machine:tick          TICK: 45, PC: 8, ADDR: 69, MEM_OUT: 104, AC: 104, DR: 104 movv ['$69', '$0']
  DEBUG   machine:tick          TICK: 46, PC: 9, ADDR: 69, MEM_OUT: 104, AC: 104, DR: 104 movv ['$69', '$1']
  DEBUG   machine:tick          TICK: 47, PC: 9, ADDR: 1, MEM_OUT: 101, AC: 104, DR: 104 movv ['$69', '$1']
  DEBUG   machine:tick          TICK: 48, PC: 9, ADDR: 1, MEM_OUT: 101, AC: 104, DR: 101 movv ['$69', '$1']
  DEBUG   machine:tick          TICK: 49, PC: 9, ADDR: 70, MEM_OUT: 0, AC: 104, DR: 101 movv ['$69', '$1']
  DEBUG   machine:tick          TICK: 50, PC: 9, ADDR: 70, MEM_OUT: 0, AC: 101, DR: 101 movv ['$69', '$1']
  DEBUG   machine:write         output: 'h' << 'e'
  DEBUG   machine:tick          TICK: 51, PC: 9, ADDR: 70, MEM_OUT: 101, AC: 101, DR: 101 movv ['$69', '$1']
  DEBUG   machine:tick          TICK: 52, PC: 10, ADDR: 70, MEM_OUT: 101, AC: 101, DR: 101 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 53, PC: 10, ADDR: 2, MEM_OUT: 108, AC: 101, DR: 101 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 54, PC: 10, ADDR: 2, MEM_OUT: 108, AC: 101, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 55, PC: 10, ADDR: 71, MEM_OUT: 0, AC: 101, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 56, PC: 10, ADDR: 71, MEM_OUT: 0, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:write         output: 'he' << 'l'
  DEBUG   machine:tick          TICK: 57, PC: 10, ADDR: 71, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 58, PC: 11, ADDR: 71, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 59, PC: 11, ADDR: 2, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 60, PC: 11, ADDR: 2, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 61, PC: 11, ADDR: 72, MEM_OUT: 0, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 62, PC: 11, ADDR: 72, MEM_OUT: 0, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:write         output: 'hel' << 'l'
  DEBUG   machine:tick          TICK: 63, PC: 11, ADDR: 72, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 64, PC: 12, ADDR: 72, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 65, PC: 12, ADDR: 3, MEM_OUT: 111, AC: 108, DR: 108 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 66, PC: 12, ADDR: 3, MEM_OUT: 111, AC: 108, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 67, PC: 12, ADDR: 73, MEM_OUT: 0, AC: 108, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 68, PC: 12, ADDR: 73, MEM_OUT: 0, AC: 111, DR: 111 movv ['$69', '$3']
  DEBUG   machine:write         output: 'hell' << 'o'
  DEBUG   machine:tick          TICK: 69, PC: 12, ADDR: 73, MEM_OUT: 111, AC: 111, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 70, PC: 13, ADDR: 73, MEM_OUT: 111, AC: 111, DR: 111 movv ['$69', '$4']
  DEBUG   machine:tick          TICK: 71, PC: 13, ADDR: 4, MEM_OUT: 32, AC: 111, DR: 111 movv ['$69', '$4']
  DEBUG   machine:tick          TICK: 72, PC: 13, ADDR: 4, MEM_OUT: 32, AC: 111, DR: 32 movv ['$69', '$4']
  DEBUG   machine:tick          TICK: 73, PC: 13, ADDR: 74, MEM_OUT: 0, AC: 111, DR: 32 movv ['$69', '$4']
  DEBUG   machine:tick          TICK: 74, PC: 13, ADDR: 74, MEM_OUT: 0, AC: 32, DR: 32 movv ['$69', '$4']
  DEBUG   machine:write         output: 'hello' << ' '
  DEBUG   machine:tick          TICK: 75, PC: 13, ADDR: 74, MEM_OUT: 32, AC: 32, DR: 32 movv ['$69', '$4']
  DEBUG   machine:tick          TICK: 76, PC: 14, ADDR: 74, MEM_OUT: 32, AC: 32, DR: 32 movv ['$69', '$5']
  DEBUG   machine:tick          TICK: 77, PC: 14, ADDR: 5, MEM_OUT: 119, AC: 32, DR: 32 movv ['$69', '$5']
  DEBUG   machine:tick          TICK: 78, PC: 14, ADDR: 5, MEM_OUT: 119, AC: 32, DR: 119 movv ['$69', '$5']
  DEBUG   machine:tick          TICK: 79, PC: 14, ADDR: 75, MEM_OUT: 0, AC: 32, DR: 119 movv ['$69', '$5']
  DEBUG   machine:tick          TICK: 80, PC: 14, ADDR: 75, MEM_OUT: 0, AC: 119, DR: 119 movv ['$69', '$5']
  DEBUG   machine:write         output: 'hello ' << 'w'
  DEBUG   machine:tick          TICK: 81, PC: 14, ADDR: 75, MEM_OUT: 119, AC: 119, DR: 119 movv ['$69', '$5']
  DEBUG   machine:tick          TICK: 82, PC: 15, ADDR: 75, MEM_OUT: 119, AC: 119, DR: 119 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 83, PC: 15, ADDR: 3, MEM_OUT: 111, AC: 119, DR: 119 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 84, PC: 15, ADDR: 3, MEM_OUT: 111, AC: 119, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 85, PC: 15, ADDR: 76, MEM_OUT: 0, AC: 119, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 86, PC: 15, ADDR: 76, MEM_OUT: 0, AC: 111, DR: 111 movv ['$69', '$3']
  DEBUG   machine:write         output: 'hello w' << 'o'
  DEBUG   machine:tick          TICK: 87, PC: 15, ADDR: 76, MEM_OUT: 111, AC: 111, DR: 111 movv ['$69', '$3']
  DEBUG   machine:tick          TICK: 88, PC: 16, ADDR: 76, MEM_OUT: 111, AC: 111, DR: 111 movv ['$69', '$6']
  DEBUG   machine:tick          TICK: 89, PC: 16, ADDR: 6, MEM_OUT: 114, AC: 111, DR: 111 movv ['$69', '$6']
  DEBUG   machine:tick          TICK: 90, PC: 16, ADDR: 6, MEM_OUT: 114, AC: 111, DR: 114 movv ['$69', '$6']
  DEBUG   machine:tick          TICK: 91, PC: 16, ADDR: 77, MEM_OUT: 0, AC: 111, DR: 114 movv ['$69', '$6']
  DEBUG   machine:tick          TICK: 92, PC: 16, ADDR: 77, MEM_OUT: 0, AC: 114, DR: 114 movv ['$69', '$6']
  DEBUG   machine:write         output: 'hello wo' << 'r'
  DEBUG   machine:tick          TICK: 93, PC: 16, ADDR: 77, MEM_OUT: 114, AC: 114, DR: 114 movv ['$69', '$6']
  DEBUG   machine:tick          TICK: 94, PC: 17, ADDR: 77, MEM_OUT: 114, AC: 114, DR: 114 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 95, PC: 17, ADDR: 2, MEM_OUT: 108, AC: 114, DR: 114 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 96, PC: 17, ADDR: 2, MEM_OUT: 108, AC: 114, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 97, PC: 17, ADDR: 78, MEM_OUT: 0, AC: 114, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 98, PC: 17, ADDR: 78, MEM_OUT: 0, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:write         output: 'hello wor' << 'l'
  DEBUG   machine:tick          TICK: 99, PC: 17, ADDR: 78, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$2']
  DEBUG   machine:tick          TICK: 100, PC: 18, ADDR: 78, MEM_OUT: 108, AC: 108, DR: 108 movv ['$69', '$7']
  DEBUG   machine:tick          TICK: 101, PC: 18, ADDR: 7, MEM_OUT: 100, AC: 108, DR: 108 movv ['$69', '$7']
  DEBUG   machine:tick          TICK: 102, PC: 18, ADDR: 7, MEM_OUT: 100, AC: 108, DR: 100 movv ['$69', '$7']
  DEBUG   machine:tick          TICK: 103, PC: 18, ADDR: 79, MEM_OUT: 0, AC: 108, DR: 100 movv ['$69', '$7']
  DEBUG   machine:tick          TICK: 104, PC: 18, ADDR: 79, MEM_OUT: 0, AC: 100, DR: 100 movv ['$69', '$7']
  DEBUG   machine:write         output: 'hello worl' << 'd'
  DEBUG   machine:tick          TICK: 105, PC: 18, ADDR: 79, MEM_OUT: 100, AC: 100, DR: 100 movv ['$69', '$7']
  DEBUG   machine:tick          TICK: 106, PC: 19, ADDR: 79, MEM_OUT: 100, AC: 100, DR: 100 halt 
  INFO    machine:simulation    output_buffer: > 'hello world'
